src/main/java/org/commcare/cases/query/queryset/CaseQuerySetLookup.java

```diff
@@ -0,0 +1,125 @@
+package org.commcare.cases.query.queryset;
+
+import org.commcare.cases.instance.CaseInstanceTreeElement;
+import org.commcare.cases.query.QueryCacheEntry;
+import org.commcare.cases.query.QueryContext;
+import org.javarosa.core.model.condition.EvaluationContext;
+import org.javarosa.core.model.instance.TreeReference;
+import org.javarosa.core.util.ArrayUtilities;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.Vector;
+
+/**
+ * Created by ctsims on 2/6/2017.
+ */
+
+public class CaseQuerySetLookup implements QuerySetLookup {
+    public static final String CASE_MODEL_ID = "case";
+
+    TreeReference caseDbRoot;
+    Map<Integer, Integer> multiplicityMap;
+
+    public CaseQuerySetLookup(TreeReference caseDbRoot, Map<Integer, Integer> multiplicityMap) {
+        this.caseDbRoot = caseDbRoot;
+        this.multiplicityMap = multiplicityMap;
+    }
+
+    @Override
+    public boolean isValid(TreeReference ref, QueryContext context) {
+        if(ref == null) {
+            return false;
+        } else {
+            ModelQuerySet set =
+                    context.getQueryCache(QuerySetCache.class).getModelQuerySet(CurrentModelQuerySet.CURRENT_QUERY_SET_ID);
+            if (set == null) {
+                //this optimization isn't useful because the current context doesn't contain
+                //a query set
+                return false;
+            }
+            return true;
```

Comment:
None of the tests in CaseDbQueryTests handle the condition where this Lookup is actually valid (and neither of the other two tests cover this code path)